optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/move/generic/PillarType.java

```diff
@@ -0,0 +1,19 @@
+package org.optaplanner.core.config.heuristic.selector.move.generic;
+
+import java.util.Comparator;
+
+public enum PillarType {
+
+    /**
+     * Pillars will only be affected in their entirety, in the order in which they're constructed.
+     */
+    FULL_ONLY,
+    /**
+     * Pillars may also be affected partially, but still only in the order in which they're constructed.
```

Comment:
"the order in which they're constructed" is ambiguous (some people could interpret it as meaning sequential).

More importantly, it's an implementation detail that we don't want to guarantee for future versions by writing it in the javadoc. It's like how OpenJDK doesn't write the actual formula for string or Integer hashcode() in the hashcode javadoc - it prohibits them from ever improving it.